third_party/java/arcs/build_defs/internal/plan.bzl

```diff
@@ -0,0 +1,83 @@
+"""Arcs Plan Generation Rules"""
+
+load("//third_party/java/arcs/build_defs/internal:kotlin.bzl", "arcs_kt_library")
+
+# Note: Once this is mature, it will replace arcs_kt_plan
+def arcs_kt_plan_2(name, package, arcs_sdk_deps, srcs = [], deps = [], visibility = None):
+    """Generates Plans Jar from protos.
+
+    Example:
+
+      ```
+      arcs_proto_plan(
+        name = "serialized_example",
+        src = "Example.arcs"
+      )
+
+      arcs_plan_generation(
+        name = "example_plan",
+        srcs = [":serialized_example"],
+        package = "com.my.example",
+      )
```

Comment:
I find this example a bit confusing. What is the difference between `arcs_proto_plan` and `arcs_plan_generation`, and how do they relate to `arcs_kt_plan_2`?

It'd probably more beneficial to elaborate in English prose to explain what the rule is for, rather than how to use it. The args block below should do a good enough job explaining what each arg does, so really all we need is a solid explanation of why you'd want to use this rule.

Reply:
I renamed the rule, and forgot to update the doc. I think seeing an example is better -- plus, we do this for other rules.

Reply:
I don't mind if you want to leave in the example, but could you please also add in some additional documentation that explains what the rule is for? At the moment all it says is "Generates Plans Jar from protos" which doesn't really tell you very much. The example is perhaps helpful for seeing how to use it, but doesn't help explain why you would want to. (Also, as we just saw, it's easy for examples to quickly become out of date.)